modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZFiniteStrain.C

```diff
@@ -25,8 +27,10 @@ ComputeAxisymmetricRZFiniteStrain::ComputeAxisymmetricRZFiniteStrain(const Input
 void
 ComputeAxisymmetricRZFiniteStrain::initialSetup()
 {
-  if (_assembly.coordSystem() != Moose::COORD_RZ)
-    mooseError("The coordinate system must be set to RZ for Axisymmetric simulations.");
+  const auto & subdomainIDs = _mesh.meshSubdomains();
```

Comment:
This is a better solution than the above ones in `computeQpResidual()`... can we do this there too?